src/indicators/drought/biomasse.py

```diff
@@ -0,0 +1,432 @@
+    anomaly.
+
+    Output is saved only if ``iso3`` is passed as
+    an argument.
+
+    Parameters
+    ----------
+    admin_pcodes: List[str]
+        List of administrative pcodes to aggregate to.
+    admin: AdminArgument
+        Admin type to subset with, one of 'ADM0', 'ADM1',
+        or 'ADM2'.
+    iso3: Union[bool, str]
+        ISO3 string or ``None``. If present, saves data
+        within that countries folder for download.
+    start_dekad: int
+        Starting dekad for annual
+        calculations.
+
+    """
+    _check_admin(admin)
+    df = load_biomasse_data(admin=admin, start_dekad=start_dekad)
+    admin_col = admin.replace("ADM", "admin") + "Pcod"
+    df_subset = df[df[admin_col].isin(admin_pcodes)]
+    # just keep to year 2000 for unique value per dekad
+    # when calculating biomasse mean aggregated
+    # since mean is the same each year
+    df_mean = df_subset[df.year.isin([2000])]
+    df_mean = (
+        df_mean[["dekad", "biomasse_mean"]]
+        .groupby("dekad")
+        .sum()
+        .reset_index()
+    )
+    df_dekad = (
+        df_subset[["year", "dekad", "biomasse"]]
+        .groupby(["year", "dekad"])
+        .sum()
+        .reset_index()
+    )
+    df_merged = pd.merge(df_dekad, df_mean, on=["dekad"])
+    df_merged["biomasse_anomaly"] = (
+        100 * df_merged["biomasse"] / df_merged["biomasse_mean"]
+    )
+
+    # sorting data again year then dekad
+    df_merged.sort_values(by=["year", "dekad"], inplace=True)
+
+    if iso3 is not None:
+        processed_filepath = _get_processed_path(
+            admin=admin, start_dekad=start_dekad, iso3=iso3
+        )
+        df_merged.to_csv(processed_filepath, index=False)
+
+    return df_merged
+
+
+def load_aggregated_biomasse_data(
+    iso3: str, admin: AdminArgument, start_dekad: int = 10
+) -> pd.DataFrame:
+    """Load aggregated Biomasse data
+
+    Parameters
+    ----------
+    iso3: str
+        ISO3 string to download data from.
+    admin: AdminArgument
+        Admin type to load, one of 'ADM0', 'ADM1',
+        or 'ADM2'.
+    start_dekad: int
+        Starting dekad for filename.
+    Returns
+    -------
+    pd.DataFrame
+    """
+    processed_filepath = _get_processed_path(
+        admin=admin, start_dekad=start_dekad, iso3=iso3
+    )
+    return pd.read_csv(processed_filepath)
+
+
+def load_biomasse_mean(mask: bool = True):
+    """Used to mask WRSI
+
+    Static file loaded from
+    http://www.geosahel.info/MetaDownload/BiomassValueMean.tif
+    so didn't bother including code to download.
+    """
+    bm_fp = _RAW_PATH / "BiomassValueMean.tif"
+    da = rioxarray.open_rasterio(bm_fp)
+    return da
+    if mask:
+        da = da.where(da.values <= 0, drop=False)
+    return da
+
+
+def _get_raw_path(admin):
+    return _RAW_PATH / _RAW_FILENAME.format(admin=admin)
+
+
+def _get_processed_path(
+    admin: AdminArgument, start_dekad: int, iso3: Union[None, str] = None
```

Comment:
No need to change, but for learning: the recommended way is to instead of using Union with None to use `Optional` see [here](https://stackoverflow.com/questions/39429526/how-to-specify-nullable-return-type-with-type-hints)

Reply:
Brilliant, thank you!!